### PR TITLE
feat: add Links field type

### LINKED ISSUE #5113: [Multi-links field] Primary Link + Secondary Links
In backend:
- add a new DOMAIN field type with "primaryLink" (string) and "secondaryLinks" (array) properties.

In frontend:
- create Domain field from settings
- display Domain field in table and show page with only the "primaryLink" for now.

### PR DESCRIPTION
**Pull request by thaisguigon**
Closes #5113

### CODE REVIEW

File: packages/twenty-server/src/engine/metadata-modules/field-metadata/composite-types/domain.composite-type.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  properties: [
+    {
+      name: 'primaryLink',
+      type: FieldMetadataType.TEXT,
+      hidden: false,
+      isRequired: true,
+    },
+    {
+      name: 'secondaryLinks',
+      type: FieldMetadataType.TEXT,
```
Comment (FelixMalfait) [inline packages/twenty-server/src/engine/metadata-modules/field-metadata/composite-types/domain.composite-type.ts:16]: should be an array?

File: packages/twenty-front/src/generated-metadata/graphql.ts [hunk 2]
```diff
@@ -286,6 +286,7 @@ export enum FieldMetadataType {
   Currency = 'CURRENCY',
   Date = 'DATE',
   DateTime = 'DATE_TIME',
+  Domain = 'DOMAIN',
```
Comment (FelixMalfait) [inline packages/twenty-front/src/generated-metadata/graphql.ts:289]: Let's rename this to LINKS and try to do something that accommodate different type of links with the options, including domain. We could have two field type DOMAINS and LINKS but it's probably not worth it

File: packages/twenty-front/src/modules/settings/data-model/fields/forms/components/SettingsDataModelFieldSettingsFormCard.tsx [hunk 3]
```diff
@@ -73,6 +73,7 @@ const previewableTypes = [
   FieldMetadataType.Text,
   FieldMetadataType.Address,
   FieldMetadataType.RawJson,
+  FieldMetadataType.Domain,
```
Comment (FelixMalfait) [inline packages/twenty-front/src/modules/settings/data-model/fields/forms/components/SettingsDataModelFieldSettingsFormCard.tsx:76]: Let's keep it behind a feature flag maybe if additionalX is not even displayed?
The goal would be to do a perfect "primary + additionalXXX" composite field (we can start with link, then phone number, emails)

File: packages/twenty-front/src/generated-metadata/graphql.ts (hunk 2 above)
Comment (thaisguigon) [reply packages/twenty-front/src/generated-metadata/graphql.ts]: Done!

File: packages/twenty-server/src/engine/metadata-modules/field-metadata/composite-types/domain.composite-type.ts (hunk 1 above)
Comment (thaisguigon) [reply packages/twenty-server/src/engine/metadata-modules/field-metadata/composite-types/domain.composite-type.ts]: As discussed with @charlesBochet, now this property is now a RAW_JSON.

**Review by lucasbordeau — APPROVED** [review]

**Pull request merged**